libsysbox/syscont/spec.go

```diff
@@ -650,17 +651,92 @@ func cfgSystemdMounts(spec *specs.Spec) {
 	spec.Mounts = append(spec.Mounts, sysboxSystemdMounts...)
 }
 
-// sysMgrSetupMounts requests the sysbox-mgr to setup special sys container mounts.
-func sysMgrSetupMounts(mgr *sysbox.Mgr, spec *specs.Spec, uidShiftRootfs bool) error {
+// Obtains the docker data-root path utilized by the inner docker process to store its
+// data. This is used to define the container mountpoint on which the host's docker
+// volume (backing this resource) will be mounted on.
+//
+// Notice that even though this code habilitates the custom definition of the data-root's
+// location, this will be only honored by Sysbox if this attribute is set prior to the
+// container creation (i.e., at docker-image build time).
+func getDockerDataRootPath(spec *specs.Spec) string {
+
+	var defaultDataRoot = "/var/lib/docker"
+
+	rootPath, err := filepath.Abs(spec.Root.Path)
+	if err != nil {
+		return defaultDataRoot
+	}
+
+	dockerCfgFile := filepath.Join(rootPath, "/etc/docker/daemon.json")
+
+	f, err := os.Open(dockerCfgFile)
+	if err != nil {
+		return defaultDataRoot
+	}
+	defer f.Close()
+
+	// Splits on newlines by default.
+	scanner := bufio.NewScanner(f)
+
+	// Parse the data-root field of the docker's config file:
+	//
+	// Example:    "data-root": "/var/lib/docker",
+	for scanner.Scan() {
+		data := scanner.Text()
+		if strings.Contains(data, "data-root") {
+
+			dataRootStr := strings.Split(data, ":")
+			if len(dataRootStr) == 2 {
+				s := dataRootStr[1]
+
+				s = strings.TrimSpace(s)
+
+				// Skip leading quotes.
+				if len(s) > 0 && s[0] == '"' {
```

Comment:
Use strings.Trim() to trim the leading & trailing quotes and comma in one shot: `strings.Trim(s, "\",")`

Reply:
Good one! Done.